pkg/generators/models/testdata/cases/objects_with_properties_and_additional_properties/expected/model_foo.go

```diff
@@ -0,0 +1,100 @@
+// This file is auto-generated, DO NOT EDIT.
+//
+// Source:
+//
+//	Title: Test
+//	Version: 0.1.0
+package generatortest
+
+import (
+	"encoding/json"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+)
+
+// Foo is an object.
+type Foo struct {
+	FooProperties
+	AdditionalProperties map[string]interface{}
+}
+
+type FooProperties struct {
+	// Bar:
+	Bar int32 `json:"bar,omitempty"`
+}
+
+// Unmarshal all named properties into FooProperties and
+// the rest into the AdditionalProperties map
+func (obj *Foo) UnmarshalJSON(data []byte) error {
+	var generic map[string]json.RawMessage
+	if err := json.Unmarshal(data, &generic); err != nil {
+		return err
+	}
+
+	obj.FooProperties = FooProperties{}
+
+	var additionalProperties = make(map[string]interface{})
+	for k, v := range generic {
+		if k == "bar" {
+			if err := json.Unmarshal(v, &(obj.FooProperties.Bar)); err != nil {
+				return err
+			}
+			continue
+		}
+		var prop interface{}
+		if err := json.Unmarshal(v, &prop); err != nil {
+			return err
+		}
+		additionalProperties[k] = prop
+	}
+
+	obj.AdditionalProperties = additionalProperties
+	return nil
+}
+
+// Marshal Foo by combining the AdditionalProperties with the
+// named properties in a single JSON object. Named properties take
+// precedence.
+func (obj Foo) MarshalJSON() ([]byte, error) {
```

Comment:
I wonder if it makes sense to add a `model_foo_test.go` like we have in https://github.com/contiamo/openapi-generator-go/blob/master/pkg/generators/models/testdata/cases/validation/expected/model_person_test.go

to validate that the MarshalJSON and UnmarshalJSON work as expected. I suspect it would be enough to unmarshal -> marshal and check that it equals the original value.

Reply:
I didn't see that one. That's a good idea. I will add a test for the json functions.